src/unicef_notification/models.py

```diff
@@ -108,16 +108,15 @@ def __str__(self):
         )
 
     def __init__(self, *args, **kwargs):
-        template_data = kwargs.pop('template_data', {})
+        template_data = kwargs.get('template_data', {})
```

Comment:
why did we cover it last time. Does anything unwanted happen if template_data is nullable value different from empty dict or the key does not exist?

I see before we're making sure the key is there with this change we're no longer ensuring, was there a reason you can think of?

Reply:
@robertavram  
from django 4 due https://code.djangoproject.com/ticket/33307#comment:4
we're experiencing this https://excubo.unicef.io/sentry/drp-prod/issues/12823/?query=is%3Aunresolved

How do you recommend to proceed?

Reply:
since the parent of this is mode.Model the super can't depend on the kwarg to be present. This will fix it. The question was about whether or not we have downstream dependencies of the kwarg to be present, because it looks intentionally added in all cases.
merging